src/services/iasql.ts

```diff
@@ -231,3 +233,99 @@ export async function dump(dbId: string, dataOnly: boolean) {
+    // It's slower, but safer to insert these records one at a time
+    for (const line of auditLogLines) {
+      await conn.query(
+        `
+        INSERT INTO iasql_audit_log (ts, "user", table_name, change_type, change, message) VALUES
+        ($1, $2, $3, $4, $5, $6);
+      `,
+        [line.ts, line.user, line.table_name, line.change_type, line.change, line.message],
+      );
+    }
+    logger.info(`Part 2 for ${db} complete!`);
+    // Restoring the `aws_account` and other modules requires the engine to be fully started
+    // We can't do that immediately, but we *can* create a polling job to do it as soon as the
+    // engine has finished starting
+    let upgradeRunning = false;
+    const upgradeHandle = setInterval(async () => {
+      const started = (await (await fetch(`http://localhost:${config.http.port}/health`)).text()) === 'ok';
+      if (!started || upgradeRunning) return;
+      logger.info(`Starting Part 3 of 3 for ${db}`);
+      upgradeRunning = true;
+      const hasCreds = existsSync(`/tmp/upgrade/${db}/creds`);
+      if (hasCreds) {
+        // Assuming the other two also exist
+        const creds = readFileSync(`/tmp/upgrade/${db}/creds`, 'utf8').trim().split(',');
+        const regionsEnabled = readFileSync(`/tmp/upgrade/${db}/regions_enabled`, 'utf8').trim().split(' ');
+        const defaultRegion = readFileSync(`/tmp/upgrade/${db}/default_region`, 'utf8').trim();
+        await conn.query(`
+          SELECT iasql_install('aws_account');
+        `);
+        await conn.query(`
+          SELECT iasql_begin();
+        `);
+        logger.info('Temporarily log the creds to see what is going on', { creds });
+        await conn.query(
+          `
+          INSERT INTO aws_credentials (access_key_id, secret_access_key) VALUES
+          ($1, $2);
+        `,
+          creds,
+        );
+        await conn.query(`
+          SELECT iasql_commit();
+        `);
+        logger.info('Regions Enabled', { regionsEnabled });
+        for (const region of regionsEnabled) {
+          await conn.query(
+            `
+            UPDATE aws_regions SET is_enabled = TRUE WHERE region = $1;
+          `,
+            [region],
+          );
+        }
+        await conn.query(
+          `
+          UPDATE aws_regions SET is_default = TRUE WHERE region = $1;
+        `,
+          [defaultRegion],
+        );
+      }
+      const moduleList = readFileSync(`/tmp/upgrade/${db}/module_list`, 'utf8').trim().split(' ');
+      logger.info('Module List', { moduleList });
+      for (const mod of moduleList) {
+        await conn.query(
+          `
+          SELECT iasql_install($1);
+        `,
+          [mod],
+        );
+      }
+      execSync(`rm -rf /tmp/upgrade/${db}`);
+      dbsDone[db] = true;
+      clearInterval(upgradeHandle);
+      logger.info(`Part 3 of 3 for ${db} complete!`);
+      if (Object.keys(dbsDone).sort().join(',') === dbs.sort().join(',')) {
```

Comment:
what should we do if they are not equal? 😨

Reply:
They usually will be not equal. They're only equal once every single DB has been upgraded.